File: backend/src/main/java/com/gestionstock/backend/service/fournisseur/CommandeService.java
package com.gestionstock.backend.service.fournisseur;

import com.gestionstock.backend.dto.fournisseur.CommandeDTO;
import com.gestionstock.backend.dto.fournisseur.CommandeRequest;
import com.gestionstock.backend.entity.enums.StatutCommande;

import java.util.List;

/**
 * Interface du service Commande.
 * Définit les opérations métier pour la gestion des commandes fournisseurs.
 */
public interface CommandeService {

    /** Récupère toutes les commandes triées par date décroissante */
    List<CommandeDTO> getAllCommandes();

    /** Récupère une commande par son identifiant (avec ses lignes) */
    CommandeDTO getCommandeById(Long id);

    /** Récupère les commandes d'un fournisseur donné */
    List<CommandeDTO> getCommandesByFournisseur(Long fournisseurId);

    /** Récupère les commandes par statut */
    List<CommandeDTO> getCommandesByStatut(StatutCommande statut);

    /** Crée une nouvelle commande avec ses lignes */
    CommandeDTO createCommande(CommandeRequest request);

    /** Met à jour le statut d'une commande (et gère le stock si LIVREE) */
    CommandeDTO updateStatut(Long id, StatutCommande statut);

    /** Supprime une commande (uniquement si EN_ATTENTE) */
    void deleteCommande(Long id);
}

File: frontend/src/app/core/services/commande.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Commande, CommandeRequest, StatutCommande } from '../models/commande.model';
import { API_BASE_URL } from './api.config';

const API_URL = API_BASE_URL;

/**
 * Service Angular pour la gestion des commandes fournisseurs.
 * Communique avec le backend via les endpoints REST /api/commandes.
 */
@Injectable({
  providedIn: 'root'
})
export class CommandeService {

  private http = inject(HttpClient);

  /** Récupère toutes les commandes */
  getAllCommandes(): Observable<Commande[]> {
    return this.http.get<Commande[]>(`${API_URL}/commandes`);
  }

  /** Récupère une commande par son ID (avec ses lignes) */
  getCommandeById(id: number): Observable<Commande> {
    return this.http.get<Commande>(`${API_URL}/commandes/${id}`);
  }

  /** Récupère les commandes d'un fournisseur */
  getCommandesByFournisseur(fournisseurId: number): Observable<Commande[]> {
    return this.http.get<Commande[]>(`${API_URL}/commandes/fournisseur/${fournisseurId}`);
  }

  /** Récupère les commandes par statut */
  getCommandesByStatut(statut: StatutCommande): Observable<Commande[]> {
    return this.http.get<Commande[]>(`${API_URL}/commandes/statut/${statut}`);
  }

  /** Crée une nouvelle commande multi-lignes */
  createCommande(request: CommandeRequest): Observable<Commande> {
    return this.http.post<Commande>(`${API_URL}/commandes`, request);
  }

  /** Met à jour le statut d'une commande */
  updateStatut(id: number, statut: StatutCommande): Observable<Commande> {
    return this.http.put<Commande>(`${API_URL}/commandes/${id}/statut?statut=${statut}`, {});
  }

  /** Supprime une commande */
  deleteCommande(id: number): Observable<void> {
    return this.http.delete<void>(`${API_URL}/commandes/${id}`);
  }
}
